mmseg/models/backbones/swin.py

```diff
@@ -601,17 +549,22 @@ def __init__(self,
                 f'The size of image should have length 1 or 2, ' \
                 f'but got {len(pretrain_img_size)}'
 
-        if isinstance(pretrained, str) or pretrained is None:
-            warnings.warn('DeprecationWarning: pretrained is a deprecated, '
+        assert not (init_cfg and pretrained), \
+            'init_cfg and pretrained cannot be specified at the same time'
+        if isinstance(pretrained, str):
+            warnings.warn('DeprecationWarning: pretrained is deprecated, '
                           'please use "init_cfg" instead')
+            self.init_cfg = dict(type='Pretrained', checkpoint=pretrained)
+        elif pretrained is None:
+            self.init_cfg = init_cfg
         else:
             raise TypeError('pretrained must be a str or None')
 
+        super(SwinTransformer, self).__init__(init_cfg=init_cfg)
```

Comment:
Should move this line to L541?
Will this override self.init_cfg (override L557)?